helix-core/src/main/java/org/apache/helix/controller/stages/ManagementMessageGenerationPhase.java

```diff
@@ -96,7 +96,9 @@ public void process(ClusterEvent event) throws Exception {
       LiveInstance liveInstance = liveInstanceMap.get(instanceName);
       Collection<Message> pendingMessages = allInstanceMessages.get(instanceName);
       String sessionId = liveInstance.getEphemeralOwner();
-      LiveInstanceStatus currentStatus = liveInstance.getStatus();
+      LiveInstanceStatus liveInstanceStatus = liveInstance.getStatus();
```

Comment:
IMHO, getStatus() returns NORMAL directly would be easier.

Reply:
It could easier, but might be confusing. If by default returning NORMAL, we cannot differentiate if the status is set to NORMAL or not set. To make the current logic consistent, I'll just keep getStatus as it is: returning NULL(not set).

Reply:
There was no convention about how a HelixProperty class should handle such a case (missing field). So we have both types of examples through our codebase. For the future, I think we should make all these different cases aligned.
1. From my perspective, each HelixProperty class should carry more business logic. And let the ZNode class to handle the raw content. So missing field can be returned from ZNode or it's methods. But should not be returned by the HelixProperty unless it is representing a specific business logic.
2. Based on point one, do you think null LiveInstanceStatus is representing any business logic? I think return NORMAL when this field is missing would be quite straightforward and is not confusing at all. Moreover, it is fully backward compatible. The older participant does not have any special status, so they will be NORMAL forever.

Reply:
Synced offline. Returning null or normal isn't a big issue. As @jiajunwang said, to simplify the business logic, I'll make the change to return NORMAL in next PR, combined with the refactoring work, making it easy to review.